core/src/mender-client.c

```diff
@@ -805,6 +801,51 @@ mender_client_initialization_work_function(void) {
     return ret;
 }
 
+static mender_err_t
+mender_commit_artifact_data(void) {
+
+    /* If provides-depends is defined, then `artifact_name` comes from the provides, else
+     * it comes from the deployment data */
```

Comment:
The code below doesn't seem to follow this comment, I think it belongs elsewhere.